### ISSUE TITLE
Different MRENCLAVEs even the same source code.

### ISSUE THREAD

**Issue by hyunsooda** [openenclave/openenclave#4062]
Hi all,

The following two dumps are from the HelloWorld example using ```oesign dump -e``` in _**between two machines**_.

From machine 1:
```
=== Entry point:
name=_start
address=0x00000000024a48

=== SGX Enclave Properties:
product_id=1
security_version=1
debug=0
xfrm=0000000000000000
num_heap_pages=1024
num_stack_pages=1024
num_tcs=1
mrenclave=33db83c4aa331d95b07c618275e0114cb2db4def4fb40380faf1157ff5129e31
```

From machine 2:
```
=== Entry point:
name=_start
address=0x00000000024a08

=== SGX Enclave Properties:
product_id=1
security_version=1
debug=0
xfrm=0000000000000000
num_heap_pages=1024
num_stack_pages=1024
num_tcs=1
mrenclave=5f00c3c056d9eea25546f90e0db8a937df5d98017356cdc09dac7e8497973fb7
```

Two MRENCLAVE should be the same ideally if the compiled code is the same including the build configuration.
As far as I know, two MRENCLAVEs should be the same since the fused hardware key is not involved in hashing the MRENCLAVE in the setup phase(EADD, EEXTEND). I guess that the ```address```  of the second row below the ```Entry point``` in the dump seems to cause this to happen. How can I set the ```address``` in the ```Entry point``` if my hypothesis is correct. Else, please let me know anything.

**Comment by anakrish** [comment]
Could it be that the build tools are different between the two machines?

**Comment by hyunsooda** [comment]
Hi @anakrish,

I found the cause of the difference. Actually, I checked the source code of ```oesign``` to see if some hardware dependency is there, but it seems not to have some relation.

The compiled objects generated by the two machines were different from each other before signing the enclave image. The static libraries were actually different. Also, generated stub files from ```oeedger8r``` were different too. that was the problem. So, I removed everything and clone & compiled them again. It now works as expected (same MRENCLAVE).

Thank you, let me close.
